shared/controller-api/src/main/java/io/pravega/shared/controller/event/UpdateReaderGroupEvent.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.controller.event;
+
+import com.google.common.collect.ImmutableSet;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+@Builder
+@Data
+@AllArgsConstructor
+public class UpdateReaderGroupEvent implements ControllerEvent {
+    private static final long serialVersionUID = 1L;
+    private final String scope;
+    private final String rgName;
+    private final long requestId;
+    private UUID readerGroupId;
+    private ImmutableSet<String> removeStreams;
```

Comment:
doesnt it also need "add streams" ?
updateRG can have new streams added and some old streams removed.

Reply:
addSubscriber() is invoked on all Streams in new config.
This takes care of adding (new) streams and updating generation for existing Streams ( present in both old and new config) 
So only streams to be removed need to be passed to the event.

Reply:
again, shouldnt we also include the generation here.. if the event is replayed we should not be unsubscribing from a stream that was added as part of a subsequent message.

Reply:
Fixed. Added generation check as suggested.